### PR TITLE
NativeAOT-LLVM: Merge nativeaotruntime subset change

### PR DESCRIPTION
**Pull request by yowl**
This PR inches forward merging in the changed subset `NativeAotBuild`/`NativeAotRuntime`

### CODE REVIEW

File: docs/workflow/building/coreclr/nativeaot.md [hunk 1]
```diff
@@ -55,7 +55,7 @@ For the compilers:
...
 - Set the enviroment variable `LLVM_CMAKE_CONFIG` to locate the LLVM config: `set LLVM_CMAKE_CONFIG=path-to-the-build-directory/lib/cmake/llvm`. This location should contain the file `LLVMConfig.cmake`. `LLVM_CMAKE_CONFIG_DEBUG` and `LLVM_CMAKE_CONFIG_RELEASE` can be used instead of `LLVM_CMAKE_CONFIG` to allow the build to select the config file based on the configuration. If set, these variables take precedence over `LLVM_CMAKE_CONFIG`.
-- Build the Jits and the ILC: `build clr.jit+clr.wasmjit+clr.nativeaotlibs+clr.tools -c [Debug|Release]`. Note that `clr.jit` only needs to be built once.  Add the `libs` subsets if you want the packages for publishing, e.g. `build clr.jit+clr.wasmjit+clr.nativeaotlibs+clr.tools+libs -c Debug`.
+- Build the Jits and the ILC: `build clr.jit+clr.wasmjit+clr.nativeaotlibs+clr.tools+clr.nativeaotruntime -c [Debug|Release]`. Note that `clr.jit` only needs to be built once.  Add the `libs` subsets if you want the packages for publishing, e.g. `build clr.jit+clr.wasmjit+clr.nativeaotlibs+clr.tools+libs -c Debug`.
```
Comment (SingleAccretion) [inline docs/workflow/building/coreclr/nativeaot.md]:
```suggestion
- Build the Jits and the ILC: `build clr.jit+clr.wasmjit+clr.nativeaotruntime+clr.nativeaotlibs+clr.tools -c [Debug|Release]`. Note that `clr.jit` only needs to be built once.  Add the `libs` subsets if you want the packages for publishing, e.g. `build clr.jit+clr.wasmjit+clr.nativeaotruntime+clr.nativeaotlibs+clr.tools+libs -c Debug`.
```

File: eng/Subsets.props [hunk 2]
```diff
@@ -315,7 +303,7 @@
     <ProjectToBuild Include="$(CoreClrProjectRoot)nativeaot\**\src\*.csproj" Category="clr" />
   </ItemGroup>
 
-  <ItemGroup Condition="$(_subset.Contains('+clr.nativeaotbuild+'))">
+  <ItemGroup Condition="$(_subset.Contains('+clr.nativeaotruntime+'))">
```
Comment (SingleAccretion) [inline eng/Subsets.props:307]: Hmm. So we're adding the build integration into native runtime build, which is not very logical, but does keep the command line length smaller. I am ok with this for the time being, but once we merge `clr.aot`, we should delete this, so there should be a `TODO-LLVM` about that.

File: eng/pipelines/runtimelab.yml [hunk 3]
```diff
@@ -84,7 +84,7 @@ stages:
         jobParameters:
           timeoutInMinutes: 300
           testGroup: innerloop
-          buildArgs: -s clr.nativeaotlibs+clr.nativeaotbuild+libs -lc Release -rc Debug
+          buildArgs: -s clr.nativeaotlibs+clr.nativeaotruntime+libs -lc Release -rc Debug
```
Comment (SingleAccretion) [inline eng/pipelines/runtimelab.yml]:
```suggestion
          buildArgs: -s clr.nativeaotruntime+clr.nativeaotlibs+libs -lc Release -rc Debug
```
Consistency.

File: eng/pipelines/runtimelab.yml [hunk 4]
```diff
@@ -142,7 +142,7 @@ stages:
         timeoutInMinutes: 300
         isOfficialBuild: ${{ variables.isOfficialBuild }}
         testGroup: innerloop
-        buildArgs: -s clr.nativeaotlibs+clr.nativeaotbuild+libs+nativeaot.packages -c Release
+        buildArgs: -s clr.nativeaotlibs+clr.nativeaotruntime+libs+nativeaot.packages -c Release
```
Comment (SingleAccretion) [inline eng/pipelines/runtimelab.yml]:
```suggestion
        buildArgs: -s clr.nativeaotruntime+clr.nativeaotlibs+libs+nativeaot.packages -c Release
```
Consistency.

File: src/coreclr/jit/phase.cpp [hunk 5]
```diff
@@ -205,10 +189,13 @@ void Phase::PostPhase(PhaseStatus status)
         PHASE_INSERT_GC_POLLS,
         PHASE_RATIONALIZE,
         PHASE_LOWERING,
-        PHASE_STACK_LEVEL_SETTER};
+        PHASE_STACK_LEVEL_SETTER
+#if TARGET_WASM
+        ,PHASE_LOWER_LLVM
+#endif
```
Comment (SingleAccretion) [inline src/coreclr/jit/phase.cpp]:
```suggestion
        PHASE_STACK_LEVEL_SETTER,
#if TARGET_WASM
        PHASE_LOWER_LLVM
#endif
```

**Comment by yowl** [comment]
Thanks!

File: src/libraries/System.Runtime/tests/System.Runtime.Tests.csproj [hunk 6]
```diff
@@ -2,7 +2,7 @@
   <PropertyGroup>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <FeatureGenericMath>false</FeatureGenericMath>
-    <NoWarn>$(NoWarn),1718,SYSLIB0013</NoWarn>
+    <NoWarn>$(NoWarn),1718,SYSLIB0013,IL3051</NoWarn>
```
Comment (SingleAccretion) [inline src/libraries/System.Runtime/tests/System.Runtime.Tests.csproj:6]: Does this need a `TODO-LLVM` to remove? Seems like a "need to merge more of upstream" kind of issue.

Comment (yowl) [reply src/libraries/System.Runtime/tests/System.Runtime.Tests.csproj:6]: I hope that's what it is.  Runs fine upstream, but the warning looks genuine and the test classes are not annotated upstream either, so not sure how it works.

Comment (SingleAccretion) [reply src/libraries/System.Runtime/tests/System.Runtime.Tests.csproj:6]: Indeed, not totally clear what is up here. Don't think there's much point in sinking time into this, though.

**Review by SingleAccretion — APPROVED** [review]
Great to see the merging process progress!

**Pull request merged**